File: java-workshop/part-2-light-mocks/src/main/java/service/GatewayShippingService.java
package service;

import domain.Order;
import domain.ShippingConfirmation;
import domain.ShippingItem;
import domain.ShippingRequest;
import gateway.ShippingGateway;

import java.util.stream.Collectors;

public class GatewayShippingService implements ShippingService {
    private final ShippingGateway shippingGateway;

    public GatewayShippingService(ShippingGateway shippingGateway) {
        this.shippingGateway = shippingGateway;
    }

    @Override
    public ShippingConfirmation createShipment(Order order) {
        ShippingRequest request = new ShippingRequest(
            order.id(),
            order.customerId(),
            order.items().stream()
                .map(item -> new ShippingItem(item.productId(), item.quantity()))
                .collect(Collectors.toList())
        );
        return shippingGateway.createShipment(request);
    }

    @Override
    public boolean cancelShipment(String trackingNumber) {
        return shippingGateway.cancelShipment(trackingNumber);
    }
}

File: csharp-workshop/Part4.Behavior/Service/GatewayShippingService.cs
using Part4.Behavior.Domain;
using Part4.Behavior.Gateway;

namespace Part4.Behavior.Service;

public class GatewayShippingService : IShippingService
{
    private readonly IShippingGateway _shippingGateway;

    public GatewayShippingService(IShippingGateway shippingGateway)
    {
        _shippingGateway = shippingGateway;
    }

    public ShippingConfirmation CreateShipment(Order order)
    {
        var request = new ShippingRequest(
            order.Id,
            order.CustomerId,
            order.Items.Select(item => new ShippingItem(item.ProductId, item.Quantity)).ToList()
        );
        return _shippingGateway.CreateShipment(request);
    }

    public bool CancelShipment(string trackingNumber)
    {
        return _shippingGateway.CancelShipment(trackingNumber);
    }
}
